src/Visitor/MutationsCollectorVisitor.php

```diff
@@ -98,15 +98,19 @@ public function leaveNode(Node $node): void
                 continue;
             }
 
-            $this->mutations[] = new Mutation(
-                $this->filePath,
-                $this->fileAst,
-                $mutator,
-                $node->getAttributes(),
-                \get_class($node),
-                $isOnFunctionSignature,
-                $isCoveredByTest
-            );
+            foreach ($mutator->mutate($node) as $mutationByMutatorIndex => $mutatedNode) {
```

Comment:
this could be like:

```
$mutations = $mutator->mutate($node);
if (!$mutations instanceof \Generator) {
   $mutations = [$mutations];
}
foreach ($mutations as $mutationByMutatorIndex => $mutatedNode) {

}
```
So you can see that these just three-four lines make everything way simpler elsewhere.

Reply:
point taken but then you can't have the return typehint and is there really any case where we care about returning a single value as opposed to an array? Is doing `return [$mutatedNode]` really more of an issue?